openmetadata-ui/src/main/resources/ui/src/components/common/DeleteWidget/DeleteWidgetModal.tsx

```diff
@@ -150,8 +151,22 @@ const DeleteWidgetModal = ({
               })
             );
 
-            if (afterDeleteAction) {
-              afterDeleteAction();
+            const shouldRunActionForDataAsset =
+              isDataAsset && entityDeleteState.softDelete;
+
+            // Execute the afterDeleteAction if passed
+            // 1. While soft deleting on the data asset details pages
+            // 2. For other pages where hard deleting does not require to go to other pages
+            //    Example: We can still show the users list after deleting one user from the table in the UsersList page
+            // Else redirect to the landing page.
+            if (
+              afterDeleteAction &&
+              (shouldRunActionForDataAsset || !isDataAsset)
+            ) {
+              // Setting timeout for a window to show the success toast
+              setTimeout(() => {
+                afterDeleteAction();
+              }, 1000);
```

Comment:
@aniketkatkar97 , why we need timeout here?

Reply:
@Sachin-chaurasiya For data assets page we passing refreshPage as afterDeleteAction, so the success toast does not appear to let user know about the result. I've added the comment for the same above that

Reply:
The comment is to show the notification.
```
// Setting timeout for a window to show the success toast
```

Reply:
I think we have added too many conditions here, @aniketkatkar97 Let's syncup for this tomorrow morning

Reply:
Sure